clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -226,7 +227,7 @@ void SYCL::fpga::BackendCompiler::ConstructJob(Compilation &C,
         types::ID Ty = getToolChain().LookupTypeForExtension(Ext.drop_front());
         if (Ty == types::TY_INVALID)
           continue;
-        if (types::isSrcFile(Ty) || Ty == types::TY_Object) {
+        if (types::isSrcFile(Ty)) {
           llvm::sys::path::replace_extension(FN, "d");
```

Comment:
Is this code still relevant?

Reply:
Yes, this is still relevant.  For the case of building all the way to final binary (instead of object) the dependency file name is formulated based on input source file and is not the recipient of a dependency unbundle step.